src/grid/grid.component.ts

```diff
@@ -0,0 +1,630 @@
+
+import {
+    DataContainer,
+    FilteringExpression,
+    IgxDialog,
+    IgxDialogModule,
+    IgxDirectivesModule,
+    SortingDirection,
+    SortingExpression,
+    StableSortingStrategy
+} from "../../src/main";
+import { DataType } from "../data-operations/data-util";
+
+
+// grid events
+
+export interface IgxGridColumnInitEvent {
+    column: IgxColumnComponent;
+}
+
+export interface IgxGridRowSelectionEvent {
+    index: number;
+    row: Object;
+}
+
+export interface IgxGridCellSelectionEvent {
+    index: number;
+    value: any;
+    row: Object;
+}
+
+export interface IgxGridFilterEvent {
+    // tBD
+}
+
+export interface IgxGridEditEvent {
+    row: Object;
+}
+
+export interface IgxGridSortEvent {
+    column: IgxColumnComponent;
+    direction: string;
+    expression: SortingExpression;
+}
+
+export interface IgxGridPagingEvent {
+    currentPage: number;
+    perPage: number;
+    totalRecords: number;
+}
+
+/**
+ *
+ *
+ * @export
+ * @class IgxGridComponent
+ * @implements {OnInit}
+ * @implements {AfterContentInit}
+ * @implements {DoCheck}
+ * @implements {OnDestroy}
+ */
+@Component({
+    moduleId: module.id,
+    selector: "igx-grid",
+    styles: [
+        `table {
+        border-collapse: collapse;
+        width: 100%;
+    }
+    table td, table th {
+        border: 1px solid #e0e0e0;
+        padding: 8px;
+    }
+
+   table tr:nth-child(even){background-color: #f5f5f5;}
+
+   table tr:hover {background-color: #e0e0e0;}
+
+   table th {
+        padding-top: 12px;
+        padding-bottom: 12px;
+        text-align: left;
+        background-color: #616161;
+        color: white;
+    }
+    .asc {
+        background-color: #aed581 !important;
+    }
+    .desc {
+        background-color: #ef5350 !important;
+    }
+    .none {
+        background-color: #616161 !important;
+    }
+    tr.selected-row td {
+        color: white;
+        background-color: #1976d2;
+    }
+    td.selected-cell {
+        color: white;
+        background-color: #29b6f6 !important;
+        outline-color: #29b6f6;
+        outline-width: 3px;
+    }
+    .selected-page {
+        color: white;
+        background-color: #29b6f6;
+    }`
+    ],
+    templateUrl: "grid.component.html",
+})
+export class IgxGridComponent implements OnInit, AfterContentInit, DoCheck, OnDestroy {
+
+    protected static nextId: number = 1;
+    @Input() public data;
+
+    @Input() get paging(): boolean {
+        return this._paging;
+    }
+
+    set paging(value: boolean) {
+        this._paging = value;
+        if (!this._paging && this.dataContainer) {
+            this.dataContainer.state.paging = null;
+        } else if (this.dataContainer) {
+            this.dataContainer.state.paging = {
+                index: 0,
+                recordsPerPage: this.rowsPerPage,
+            };
+        }
+        this.shouldDataBind = true;
+    }
+
+    @Input() public rowsPerPage: number = 25;
```

Comment:
Stick to the same naming. If in the event the `rowsPerPage` is called `perPage`, then either change it here, or change it in the event.